### PR TITLE
[DeadCode] Add RemoveJustVariableAssignRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: rules/PSR4/Rector/FileWithoutNamespace/NormalizeNamespaceByPSR4ComposerAutoloadRector.php [hunk 1]
```diff
@@ -148,8 +149,7 @@ private function refactorFileWithoutNamespace(
             }
         }
 
-        $namespace = new Namespace_(new Name($expectedNamespace), $nodes);
-        $nodesWithStrictTypesThenNamespace[] = $namespace;
+        $nodesWithStrictTypesThenNamespace[] = new Namespace_(new Name($expectedNamespace), $nodes);
```
Comment (samsonasik) [inline rules/PSR4/Rector/FileWithoutNamespace/NormalizeNamespaceByPSR4ComposerAutoloadRector.php]: `$namespace` is used on next node after `if` in line 157, so `ExprUsedInNextNodeAnalyzer->isUsed()` check should be used.

File: packages/Testing/PHPUnit/AbstractRectorTestCase.php [hunk 2]
```diff
@@ -123,7 +123,7 @@ protected function getFixtureTempDirectory(): string
     private function doTestFileMatchesExpectedContent(
         SmartFileInfo $originalFileInfo,
         SmartFileInfo $expectedFileInfo,
-        SmartFileInfo $fixtureFileInfo,
+        \SmartFileInfo $fixtureFileInfo,
```
Comment (samsonasik) [inline packages/Testing/PHPUnit/AbstractRectorTestCase.php]: there is no `\SmartFileInfo`

File: rules/PSR4/Rector/FileWithoutNamespace/NormalizeNamespaceByPSR4ComposerAutoloadRector.php (hunk 1 above)
Comment (TomasVotruba) [reply rules/PSR4/Rector/FileWithoutNamespace/NormalizeNamespaceByPSR4ComposerAutoloadRector.php]: I'll fix the CI first :). I'll ask you for review after :+1:

File: rules/Privatization/Reflection/ClassConstantsResolver.php [hunk 3]
```diff
@@ -34,8 +35,7 @@ public function getClassConstantNamesToValues(string $classWithConstants): array
         $classReflection = $this->reflectionProvider->getClass($classWithConstants);
         $nativeReflection = $classReflection->getNativeReflection();
 
-        $constantNamesToValues = $nativeReflection->getConstants();
-        $this->cachedConstantNamesToValues = $constantNamesToValues;
+        $this->cachedConstantNamesToValues = $nativeReflection->getConstants();
```
Comment (samsonasik) [inline rules/Privatization/Reflection/ClassConstantsResolver.php]: `$constantNamesToValues` is used after assigned, so `ExprUsedInNextNodeAnalyzer->isUsed($expr)` check is needed.

Comment (TomasVotruba) [reply rules/Privatization/Reflection/ClassConstantsResolver.php]: I'll fix the CI first :). I'll ask you for review after. Not neccessary to comment now, the CI is showing the bugs.

**Comment by TomasVotruba** [comment]
Moved to https://github.com/rectorphp/rector-src/pull/2618 due to stucked CI

**Pull request closed without merging**